platform/viewer/src/App.js

```diff
@@ -104,31 +119,50 @@ class App extends Component {
   constructor(props) {
     super(props);
 
-    this._appConfig = props;
+    const { config, defaultExtensions } = props;
 
-    const { servers, extensions, hotkeys, oidc } = props;
+    const appDefaultConfig = {
+      tools: {},
+      extensions: [],
+      routerBasename: '/',
+      whiteLabelling: {},
+    };
+
+    this._appConfig = {
+      ...appDefaultConfig,
+      ...(typeof config === 'function' ? config({ servicesManager }) : config),
+    };
+
+    const { servers, hotkeys, tools, extensions, oidc } = this._appConfig;
 
     this.initUserManager(oidc);
-    _initServices([UINotificationService, UIModalService, UIDialogService]);
-    _initExtensions(extensions, hotkeys);
+    _initServices([
+      UINotificationService,
+      UIModalService,
+      UIDialogService,
+      UIContextMenuService,
+      UILabellingFlowService,
+    ]);
+    _initExtensions([...defaultExtensions, ...extensions], tools);
```

Comment:
Okay. I see why you did this. We are tightly coupled to `cornerstone` extension, so it's _always_ baked in. That makes it difficult to provide it's configuration in the `extensions` array.

Having a specific spot for it in the application config stinks, but it's our only option until we decouple.

My only ask is that we switch from providing something as specific as `tools`, and just pass in an entire `cornerstoneExtensionConfig` that has a `tools` key, that way we won't have a breaking change as we add more values we'd like to set.

Reply:
Updated.